src/full/Agda/Syntax/Concrete/Definitions/Types.hs

```diff
@@ -158,16 +171,16 @@ isInterleavedData :: InterleavedDecl -> Maybe ()
 isInterleavedData InterleavedData{} = Just ()
 isInterleavedData _ = Nothing
 
-interleavedDecl :: Name -> InterleavedDecl -> [(Int, NiceDeclaration)]
+interleavedDecl :: Name -> InterleavedDecl -> [(DeclNum, NiceDeclaration)]
 interleavedDecl k = \case
-  InterleavedData (i, d@(NiceDataSig _ acc abs pc uc _ pars _)) ds ->
+  InterleavedData i d@(NiceDataSig _ acc abs pc uc _ pars _) ds ->
     let fpars = concatMap dropTypeAndModality pars
         ddef  = NiceDataDef noRange UserWritten abs pc uc k fpars
-    in (i,d) : maybe [] (\ (j, dss) -> [(j, ddef (concat (reverse dss)))]) ds
-  InterleavedFun (i, d@(FunSig r acc abs inst mac info tc cc n e)) dcs ->
-    let fdef dcss = let (dss, css) = unzip dcss in
-                    FunDef r (concat dss) abs inst tc cc n (concat css)
-    in (i,d) : maybe [] (\ (j, dcss) -> [(j, fdef (reverse dcss))]) dcs
+    in (i,d) : maybe [] (\ (j, dss) -> [(j, ddef (sconcat (List1.reverse dss)))]) ds
```

Comment:
One of these days we really should introduce snoclists so that we stop confusing reversing a
list & turning a push-to-the-end one into its push-to-the-front equivalent.